### PR TITLE
[22.05] unbound: fix CVE-2022-30698 and CVE-2022-30699

### PR DESCRIPTION
**Pull request by ajs124**
###### Description of changes
https://lists.nlnetlabs.nl/pipermail/unbound-users/2022-August/007851.html

###### Things done
- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/networking/unbound/default.nix [hunk 1]
```diff
@@ -107,6 +108,15 @@ stdenv.mkDerivation rec {
 
   doCheck = true;
 
+  patches = [
+    # Fix CVE-2022-30698 and CVE-2022-30699
+    (fetchpatch {
+      url = "https://github.com/NLnetLabs/unbound/commit/f6753a0f1018133df552347a199e0362fc1dac68.patch";
```
Comment (risicle) [inline pkgs/tools/networking/unbound/default.nix:114]:
```suggestion
      name = "CVE-2022-30698.CVE-2022-30699.patch";
      url = "https://github.com/NLnetLabs/unbound/commit/f6753a0f1018133df552347a199e0362fc1dac68.patch";
```

Give vulnix a chance at detecting the fix.

**Comment by risicle** [comment]
It occurs to me that `pyunbound` inherits the `src` of `unbound` but not the patches. So it doesn't get any of the fixes we apply like this - note this doesn't cause a rebuild of `pyunbound`.

**Comment by risicle** [comment]
(should probably push that upstream too)

**Comment by fpletz** [comment]
IMHO we also can just bump unbound like on master. Looking at the changelog, I don't think we can expect any breakage.

**Comment by ajs124** [comment]
The changelogs do mention at least one backwards incompatibility, e.g. https://www.nlnetlabs.nl/news/2022/Feb/10/unbound-1.15.0-released/ says "The default for the aggressive-nsec option has changed"

**Comment by fpletz** [comment]
@ajs124 You're right, sorry. Didn't notice the version in 22.05 was that old.

**Comment by risicle** [comment]
Checking out the patch, I'm satisfied to some extent by us running the tests and this change requiring a test change which presumably means the tests at least run the affected code.

**Comment by ajs124** [comment]
@vcunat mentioned on matrix that some people seem not to agree that this is a security issue at all. I'm not a DNS expert, I just run DNS servers, so I'd personally defer to the developers of the software that I run, but if any of you are experts in this and have a strong opinion on this, I'm happy to hear them.

**Comment by vcunat** [comment]
For context, the security researchers marked this a vulnerability because it allows (partially) evading domain take-downs (which could be done e.g. because malware is distributed from there or it's used as C&C).  The usual counter-argument is that malicious actors can register their names at places that won't do take-downs.

**Pull request merged**